lib/jxl/render_pipeline/simple_render_pipeline.cc

```diff
@@ -188,6 +189,59 @@ void SimpleRenderPipeline::ProcessBuffers(size_t group_id, size_t thread_id) {
           channel_data_[c],
           Rect(kRenderPipelineXOffset, kRenderPipelineXOffset, xsize, ysize));
     }
+
+    if (stage->SwitchToImageDimensions()) {
+      size_t image_xsize, image_ysize;
+      FrameOrigin frame_origin;
+      stage->GetImageDimensions(&image_xsize, &image_ysize, &frame_origin);
+      frame_dimensions_.Set(image_xsize, image_ysize, 0, 0, 0, false, 1);
+      std::vector<ImageF> old_channels = std::move(channel_data_);
+      channel_data_.clear();
+      channel_data_.reserve(old_channels.size());
+      for (size_t c = 0; c < old_channels.size(); c++) {
+        channel_data_.emplace_back(2 * kRenderPipelineXOffset + image_xsize,
+                                   2 * kRenderPipelineXOffset + image_ysize);
+      }
+      for (size_t y = 0; y < image_ysize; ++y) {
+        for (size_t c = 0; c < channel_data_.size(); c++) {
+          output_rows[c].resize(1);
+          output_rows[c][0] = channel_data_[c].Row(kRenderPipelineXOffset + y);
+        }
+        // TODO(sboukortt): consider doing this only on the parts of the
```

Comment:
If this is just a question of optimization, then the slow pipeline doesn't need to care.
If this is required by the stage, then yes :D

Reply:
Then no need to keep the todo around :)